msgraph-developer-proxy/ProxyEngine.cs

```diff
@@ -25,15 +25,16 @@ public class ProxyEngine {
     // used for deciding which URLs to decrypt for further inspection
     private ISet<Regex> _hostsToWatch = new HashSet<Regex>();
 
-    private static string __productVersion = string.Empty;
-    private static string _productVersion {
+    private static string _productVersion = string.Empty;
+    public static string ProductVersion {
         get {
-            if (__productVersion == string.Empty) {
-                var fileVersionInfo = FileVersionInfo.GetVersionInfo(AppContext.BaseDirectory);
-                __productVersion = fileVersionInfo?.ProductVersion!;
+            if (_productVersion == string.Empty) {
+                var assemblyPath = typeof(ProxyEngine).Assembly.Location;
```

Comment:
`Assembly.Location` doesn't work in self contained mode and was the source of our issues with plugin loading:
warning IL3000: 'System.Reflection.Assembly.Location' always returns an empty string for assemblies embedded in a single-file app. If the path to the app directory is needed, consider calling 'System.AppContext.BaseDirectory'.

Reply:
Right. We need to find a way to get the executable name as well on top of its location.

Is there btw a way to use the same type of build locally as we use in the workflow to avoid different behaviors and catch such discrepancies earlier?

Reply:
from the CLI absolutely! which means you can probably configure it in VSCode. This is building a release build, but changing the -c option to Debug should work just fine

```bash
 dotnet build ./msgraph-developer-proxy/msgraph-developer-proxy.csproj -c Release -r osx-x64 --self-contained
 dotnet build ./msgraph-developer-proxy-plugins/msgraph-developer-proxy-plugins.csproj -c Release -r osx-x64 --no-self-contained
```

Reply:
Regarding the file path to the executable, perhaps this helps: https://github.com/dotnet/runtime/issues/13051#issuecomment-510267727

Reply:
It's odd, but when I use the build statements you shared above locally, I can get the assembly path just fine:

<img width="566" alt="image" src="https://user-images.githubusercontent.com/11164679/216267901-c9fe7413-3cb1-4e85-8660-55914bd011db.png">

<img width="637" alt="image" src="https://user-images.githubusercontent.com/11164679/216267818-4686ca34-0d7a-4b05-a75b-6e0d8fa2fbc8.png">

I tried both Release and Debug mode to see if it makes a difference, but in both cases the path is displayed just fine. What am I missing?

Reply:
That is really odd. I'll take a look and see if I can reproduce it.

Reply:
If we add 
```
<PublishSingleFile>true</PublishSingleFile>
```

to `msgraph-developer-proxy.csproj` as a child of the first `<PropertyGroup>` element then in full VS we'll get better hints that about the Assembly.Location issue:
![image](https://user-images.githubusercontent.com/7122716/217068527-06ba04e7-b8d3-4eda-be5b-b2e2ce007b39.png)

Reply:
How about we make this change?

```suggestion
                var assemblyPath = Process.GetCurrentProcess()?.MainModule?.FileName ?? typeof(ProxyEngine).Assembly.Location;
```